### DISCUSSION TITLE
Suggestion: Switch default playback device (and/or profiles) by hotkey

### DISCUSSION THREAD

**Discussion by AquaeAtrae** [File-New-Project/EarTrumpet#659] [Ideas]
Thanks for this useful and much needed app! Great work.

I was hoping the "hotkey" I read about would help quickly switch between a gaming headset, a VR headset, and speakers (for movies, etc). I was disappointed that it just brings up the mixer and that the mixer cannot switch default devices. As is, this common switch requires multiple clicks with either EarTrumpet or the native Windows control. Many keyboards and headsets have mappable buttons that would be great for this if only hotkeys existed.

For many, a single hotkey (say Ctrl-Shift-`) simply needs to "switch to last playback/recording devices". But for more complicated setups like my own, additional hotkeys could be fixed to each device (Ctrl-Shift-1 for Speakers, Ctrl-Shift-2 for Gaming Headset, Ctrl-Shift-3 for VR Headset). 

If and when you're ready to expand upon the idea, these hotkeys could one day "load" different profiles rather than just setting the default devices. A profile would "save" not only the default devices but also different volume levels for those devices and mixed levels and device assignments for specific apps. This is more than I require, but I could definitely see the value for mine and other situations.

**Comment by riverar** [comment]
Yup got it!

**Comment by keepitsane** [comment]
Just wondering if there was any updates or on this feature request? Thanks you in advance!

**Comment by riverar** [comment]
Hey @keepitsane, we have an EarTrumpet addon in the pipeline and this feature was completed as part of that effort. Will push to get that out soon. I'll ping you two for some testing, if you're up to it.

**Comment by keepitsane** [comment]
@riverar For sure would love to help out in anyway possible. Thank for developing this tool, it is very userful!

**Comment by LoungeFlyZ** [comment]
@riverar would love to beta test this too if you need more people!

**Comment by TheXenocide** [comment]
So I'm not sure what the add-on you're working on is doing, but I find myself in a situation where I have 3 very different sound setups (normal usage, video chat, streaming). I'm really hopeful this could control both input devices and output devices on a per-app basis with an easy profile switch that isn't just the global system output, but a full set of app-specific settings. For example:

Profile 1:
- Default:
  - Playback: Desktop Speakers
  - Communication Playback: Desktop Speakers
  - Recording: RealTekAudio USB (Mute)
  - Communication Recording: RealTek USB (Mute)
- All Apps set to Default Playback
- All Apps set to Default Recording

Profile 2:
- Default:
  - Playback: Desktop Speakers
  - Communication Playback: Razer Audio Controller - Chat
  - Recording: Razer Audio Controller - Chat
  - Recording Communication: VB-Audio Cable D
- No Man's Sky: 
  - Playback: VB-Audio Cable A
- Discord: 
  - Playback: VB-Audio Cable B
  - Recording: VB-Audio Cable C
- etc.

Basically I'm using multiple virtual audio devices to provide different mixes to different output sources (recording, stream, chat, etc.). It's really a lot of work to switch back and forth between all of them and it's very difficult to verify it's all setup right (it's easy to forget one little setting somewhere and boom no microphone going to discord). I don't personally need a hotkey, but it seems like it would go well with it, but if it was possible to create profiles like these and switch a bunch of apps/settings all at once that would save me so much time/effort. I'd be happy to donate a few bucks for the pain it would save me. If this issue isn't the right place for it I can always open a separate feature request.

**Reply by Jojoshua** [reply]
Hi @riverar , I have installed the DEV version to test this feature. It seems to work well but I was hoping to find a specific type of action that I did not see available. This would be to redirect the audio of a specific app to a specific playback device. For example I want to set ctrl + numpad1 to move all Chrome audio to headphone and ctrl + numpad2 to move it back to speakers. Is this currently possible?

**Reply by AquaeAtrae** [reply]
I suspect per-app routing may be more complicated than simply manipulating existing Windows' settings, although the devs would know better than I. That would be very useful if possible.

Thanks though for noting the coming feature in the dev channel (https://install.eartrumpet.app/dev/EarTrumpet.Package.appinstaller). Looks like Actions can "Set the default playback device", but not yet "Set the default communication device". I'll play with it more. I look forward to simply being able to switch to/from headset audio from the wireless headset's (Logitech G935) own buttons. :)

https://i.imgur.com/sKihSVY.png

**Reply by bunch92** [reply]
Thx for that dev version , it works great for switching playback devices using shortcuts

**Comment by vertigo220** [comment]
Is this still planned?

**Comment by IsaacInsoll** [comment]
I'd love a keyboard shortcut just to toggle between outputs. I only have two: headphones and speakers, and having a keyboard shotcut to switch between them, rather than right clicking and choosing the other one, would be awesome

**Comment by NotGovernor** [comment]
Checking on the status on this. Something we can use today? +1 if not.
